src/AutoRest.CSharp/Common/Input/InputTypes/InputUnionType.cs

```diff
@@ -13,7 +12,7 @@ internal record InputUnionType(string Name, IReadOnlyList<InputType> UnionItemTy
     internal IReadOnlyList<InputEnumTypeValue> GetEnum()
     {
         if (!IsAllLiteralStringPlusString())
-            throw new InvalidOperationException($"Cannot convert union '{Name}' to enum because its not all strings");
+            throw new InvalidOperationException($"Cannot convert union '{this}' to enum because its not all strings");
```

Comment:
Can we refactor this error message for 'because its not all strings'?
Also, do we throw exception directly here?  Seems this is still a valid typespec.

Reply:
we always check `IsAllLiteralStringPlusString` before we call this method therefore the exception will never throw.

Change to this because this.ToString() contains a lot more information than Name. The name of a union will always be Union which is just useless